gpt_index/query_engine/router_query_engine.py

```diff
@@ -0,0 +1,50 @@
+import logging
+from typing import Optional, Sequence
+from gpt_index.indices.query.base import BaseQueryEngine
+from gpt_index.indices.query.schema import QueryBundle
+from gpt_index.query_engine.types import QueryEngineWithMetadata
+from gpt_index.response.schema import RESPONSE_TYPE
+from gpt_index.selectors.llm_selectors import LLMSingleSelector
+from gpt_index.selectors.types import BaseSelector
+
+logger = logging.getLogger(__name__)
+
+
+class RouterQueryEngine(BaseQueryEngine):
+    def __init__(
+        self,
+        selector: BaseSelector,
+        query_engines_with_metadata: Sequence[QueryEngineWithMetadata],
```

Comment:
Hm, this seems a bit hard to use by the user, as in they would have to import a new type `from gpt_index.query_engine.types import QueryEngineWithMetadata` just to use the query engine.

A few points:
- If we do go with this approach, then in general it'd be good to make any user-facing module easily improtable. `from gpt_index.query_engine import QueryEngineWithMetadata` shorter/better than current
- same thing for Metadata (since that's also defined by user) 
- also should add docs for the above

- have you thought about unifying this with a more general Tool abstraction? e.g. at a basic level a Tool can take in a description and a callable that's called when the Tool is used. Could be helpful if we're building more general tool abstractions later

Reply:
actually would be kinda cool to have a converter function from QueryEngineWithMetadata, so it can easily be used as a tool in a langchain agent etc.

Reply:
revised to `QueryEngineTool`